.../include/ext2fs/ext2fs.h:1402:39: error: unknown type name 'dev_t'; did you mean 'div_t'?
 1402 | extern char *ext2fs_find_block_device(dev_t device);
      |                                       ^~~~~
      |                                       div_t
.../include/ext2fs/ext2fs.h:1822:62: error: unknown type name 'mode_t'
 1822 | extern int ext2fs_open_file(const char *pathname, int flags, mode_t mode);
      |  

---

--- a/src/block.c
+++ b/src/block.c
@@ -22,6 +22,10 @@
 #include <string.h>
 #include <unistd.h>
 
+#ifndef __GLIBC__
+#include <sys/stat.h>
+#endif /* musl */
+
 #ifndef EXT2_FLAT_INCLUDES
 #define EXT2_FLAT_INCLUDES 0
 #endif
--- a/src/file_type.c
+++ b/src/file_type.c
@@ -23,6 +23,10 @@
 #include <ctype.h>
 #include <string.h>
 
+#ifndef __GLIBC__
+#include <sys/stat.h>
+#endif /* musl */
+
 #include "util.h"
 #include "inode.h"
 
--- a/src/hard_link_stack.c
+++ b/src/hard_link_stack.c
@@ -22,6 +22,10 @@
 #include <unistd.h>
 #include <stdlib.h>
 
+#ifndef __GLIBC__
+#include <sys/stat.h>
+#endif /* musl */
+
 #include "hard_link_stack.h"
 
 static struct link_stack_head head;
--- a/src/inode.h
+++ b/src/inode.h
@@ -18,6 +18,9 @@
 #ifndef INODE_H
 #define INODE_H
 
+#ifndef __GLIBC__
+#include <sys/stat.h>
+#endif /* musl */
 
 #include <ext2fs/ext2fs.h>
 #include "util.h"
--- a/src/lookup_local.c
+++ b/src/lookup_local.c
@@ -23,6 +23,10 @@
 #include <unistd.h>
 #endif
 
+#ifndef __GLIBC__
+#include <sys/stat.h>
+#endif /* musl */
+
 #include <ext2fs/ext2fs.h>
 //#include <ext2fs/ext2_io.h>
 #include "ext2fsP.h"
--- a/src/ring_buf.h
+++ b/src/ring_buf.h
@@ -20,6 +20,10 @@
 #ifndef RING_BUF_H
 #define RING_BUF_H
 
+#ifndef __GLIBC__
+#include <sys/stat.h>
+#endif /* musl */
+
 /* ext3/4 libraries */
 #include <ext2fs/ext2fs.h>
 
--- a/src/util.c
+++ b/src/util.c
@@ -22,6 +22,10 @@
 #include <string.h>
 #include <time.h> 
 
+#ifndef __GLIBC__
+#include <sys/stat.h>
+#endif /* musl */
+
 /* ext3/4 libraries */
 #include <ext2fs/ext2fs.h>
 #include <e2p/e2p.h>
